File: 백준/Silver/11403. 경로 찾기/경로 찾기.py
import sys
input = sys.stdin.readline

n = int(input())

graph = [list(map(int, input().rstrip().split())) for _ in range(n)]

for k in range(n):
    for a in range(n):
        for b in range(n):
            if graph[a][k] == 1 and graph[k][b] == 1:
                graph[a][b] = 1
                
for a in range(n):
    for b in range(n):
        print(graph[a][b], end=' ')
        
    print()

File: 백준/Silver/11403. 경로 찾기/경로 찾기.java
import java.io.*;

public class Main {
	public static int[][] graph;
	public static int N;

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringBuilder sb = new StringBuilder();

		N = Integer.parseInt(br.readLine());
		graph = new int[N][N];

		for (int i = 0; i < N; i++) {
			String[] row = br.readLine().split(" ");
			for (int j = 0; j < N; j++) {
				graph[i][j] = Integer.parseInt(row[j]);
			}
		}

		for (int k = 0; k < N; k++) {
			for (int a = 0; a < N; a++) {
				for (int b = 0; b < N; b++) {
					if (graph[a][k] == 1 && graph[k][b] == 1) {
						graph[a][b] = 1;
					}
				}
			}
		}

		for (int i = 0; i < N; i++) {
			for (int j = 0; j < N; j++) {
				sb.append(graph[i][j] + " ");
			}
			sb.append("\n");
		}

		System.out.println(sb);

	}

}